http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -461,4 +535,89 @@ private void addScopes(String scopes, Set<String> allScopes) {
+        jwtClaims.setAudience(deployment.getIssuerUrl());
+
+        for ( NameValuePair parameter: forwardedQueryParams) {
+            jwtClaims.setClaim(parameter.getName(), parameter.getValue());
+        }
+        jwtClaims.setClaim(STATE, state);
+        jwtClaims.setClaim(REDIRECT_URI, redirectUri);
+        jwtClaims.setClaim(RESPONSE_TYPE, CODE);
+        jwtClaims.setClaim(CLIENT_ID, deployment.getResourceName());
+
+        // sign JWT first before encrypting
+        JsonWebSignature signedRequest = signRequest(jwtClaims, deployment);
+
+        // Encrypting optional
+        if (deployment.getRequestObjectEncryptionAlgValue() != null && !deployment.getRequestObjectEncryptionAlgValue().isEmpty() &&
+            deployment.getRequestObjectEncryptionEncValue() != null && !deployment.getRequestObjectEncryptionEncValue().isEmpty()) {
+            return encryptRequest(signedRequest).getCompactSerialization();
+        } else {
+            return signedRequest.getCompactSerialization();
+        }
+    }
+
+    private static KeyPair getkeyPair(OidcClientConfiguration deployment) throws IOException {
+        if (deployment.getRequestObjectSigningKeyStoreFile().contains(PROTOCOL_CLASSPATH)) {
+            deployment.setRequestObjectSigningKeyStoreFile(deployment.getRequestObjectSigningKeyStoreFile().replace(PROTOCOL_CLASSPATH, Objects.requireNonNull(Thread.currentThread().getContextClassLoader().getResource("")).getPath()));
+        }
+        if (!deployment.getRequestObjectSigningAlgorithm().equals(NONE) && deployment.getRequestObjectSigningKeyStoreFile() == null){
+            throw log.invalidKeyStoreConfiguration();
+        } else {
+            return JWTSigningUtils.loadKeyPairFromKeyStore(deployment.getRequestObjectSigningKeyStoreFile(),
+                    deployment.getRequestObjectSigningKeyStorePassword(), deployment.getRequestObjectSigningKeyPassword(),
+                    deployment.getRequestObjectSigningKeyAlias(), deployment.getRequestObjectSigningKeyStoreType());
+        }
+    }
+
+    private static JsonWebSignature signRequest(JwtClaims jwtClaims, OidcClientConfiguration deployment) throws IOException, JoseException {
+        JsonWebSignature jsonWebSignature = new JsonWebSignature();
+        jsonWebSignature.setPayload(jwtClaims.toJson());
+
+        if (!deployment.getRequestObjectSigningAlgValuesSupported().contains(deployment.getRequestObjectSigningAlgorithm())) {
+            throw log.invalidRequestObjectSignatureAlgorithm();
+        } else {
+            if (deployment.getRequestObjectSigningAlgorithm().contains(NONE)) { //unsigned
+                jsonWebSignature.setAlgorithmConstraints(AlgorithmConstraints.NO_CONSTRAINTS);
+                jsonWebSignature.setAlgorithmHeaderValue(NONE);
+            } else if (deployment.getRequestObjectSigningAlgorithm().contains(HMAC_SHA256)
+                    || deployment.getRequestObjectSigningAlgorithm().contains(HMAC_SHA384)
```

Comment:
Any reason we are using .contains instead of .equals for the comparisons here (e.g., AlgorithmIdentifiers.HMAC_SHA384.equals(...))

Reply:
fixed